packages/e2e/test/wallet/PersonalWallet/conwayTransactions.test.ts

```diff
@@ -0,0 +1,395 @@
+import { PersonalWallet } from '@cardano-sdk/wallet';
+import { logger } from '@cardano-sdk/util-dev';
+
+import { firstValueFrom, map } from 'rxjs';
+import { getEnv, getWallet, submitAndConfirm, walletReady, walletVariables } from '../../../src';
+
+/*
+Use cases not covered by specific tests because covered by (before|after)(All|Each) hooks
+- send RegisterDelegateRepresentative certificate
+  sent in beforeAll as a registered DRep is required to some other test cases
+- send UnregisterDelegateRepresentative certificate
+  sent in afterAll as test suite tear down
+- send Conway Registration certificate
+  sent in beforeEach in nested describe with tests requiring it
+- send ProposalProcedure
+  sent in beforeAll in nested describe to check VotingProcedure
+- send AuthorizeCommitteeHot
+  sent in beforeAll in nested describe to check UpdateDelegateRepresentative
+*/
+
+const {
+  CertificateType,
+  CredentialType,
+  GovernanceActionType,
+  RewardAccount,
+  StakeKeyStatus,
+  StakePoolStatus,
+  Vote,
+  VoterType
+} = Cardano;
+
+const env = getEnv(walletVariables);
+
+const anchor = {
+  dataHash: '3e33018e8293d319ef5b3ac72366dd28006bd315b715f7e7cfcbd3004129b80d' as Crypto.Hash32ByteBase16,
+  url: 'https://testing.this'
+};
+
+const assertTxHasCertificate = (tx: Cardano.HydratedTx, certificate: Cardano.Certificate) =>
+  expect(tx.body.certificates![0]).toEqual(certificate);
+
+const getTestWallet = async (idx: number, name: string, minCoinBalance?: bigint) => {
+  const { wallet } = await getWallet({ env, idx, logger, name, polling: { interval: 50 } });
+
+  await walletReady(wallet, minCoinBalance);
+
+  return wallet;
+};
+
+describe('PersonalWallet/conwayTransactions', () => {
+  let dRepWallet: PersonalWallet;
+  let wallet: PersonalWallet;
+
+  let dRepCredential: Cardano.Credential & { type: Cardano.CredentialType.KeyHash };
+  let poolId: Cardano.PoolId;
+  let rewardAccount: Cardano.RewardAccount;
+  let stakeCredential: Cardano.Credential;
+
+  let dRepDeposit: bigint;
+  let governanceActionDeposit: bigint;
+  let stakeKeyDeposit: bigint;
+
+  const assertWalletIsDelegating = async () =>
+    expect((await firstValueFrom(wallet.delegation.rewardAccounts$))[0].delegatee?.nextNextEpoch?.id).toEqual(poolId);
+
+  const cleanWallet = async () => {
+    const rewardAccounts = await firstValueFrom(wallet.delegation.rewardAccounts$);
+
+    if (!rewardAccounts.some((acct) => acct.keyStatus === StakeKeyStatus.Unregistered)) {
+      const { tx } = await wallet.createTxBuilder().delegatePortfolio(null).build().sign();
+
+      await submitAndConfirm(wallet, tx, 1);
+    }
+  };
+
+  const getDRepCredential = async () => {
+    const drepPubKey = await dRepWallet.getPubDRepKey();
+    const dRepKeyHash = Crypto.Hash28ByteBase16.fromEd25519KeyHashHex(
+      (await Crypto.Ed25519PublicKey.fromHex(drepPubKey).hash()).hex()
+    );
+
+    dRepCredential = { hash: dRepKeyHash, type: CredentialType.KeyHash };
```

Comment:
Just nitpicking, but I would prefer to have these get* functions return the values, rather than updating an upper scope variable

Reply:
Or better, rename them to `set*`. They are only used in the before* statements to initialize state, so it would make sense to be setters.